### PR TITLE
Very descriptive comment on usage of deploy function

### PR DESCRIPTION
**Pull request by jugglingjsons**
Q1: https://github.com/serverless/serverless/issues/8752

Based on discussion here https://github.com/serverless/serverless/issues/8752
Basically when you have provisionedConcurrency set you can be confused as to why your lambdas are not updating after you run this command.
Looking for pointers where it belongs if not here

Closes: #{8752}

### CODE REVIEW

**Comment by jugglingjsons** [comment]
Not sure if this still counts as docs, but happy to put it anywhere else it needs to go :)

**Comment by pgrzesik** [comment]
Hello @jugglingjsons, thanks for submitting this proposal. I believe we should include this type of information in docs (see https://github.com/serverless/serverless/blob/master/docs/providers/aws/cli-reference/deploy-function.md and https://github.com/serverless/serverless/blob/master/docs/providers/aws/guide/deploying.md) instead of including longer description in `usage`. What do you think?

**Comment by jugglingjsons** [comment]
> Hello @jugglingjsons, thanks for submitting this proposal. I believe we should include this type of information in docs (see https://github.com/serverless/serverless/blob/master/docs/providers/aws/cli-reference/deploy-function.md and https://github.com/serverless/serverless/blob/master/docs/providers/aws/guide/deploying.md) instead of including longer description in `usage`. What do you think?

Sure, agreed! But I believe there should be at least a short-link to this more descriptive explanation in the cli. WDYT?

**Comment by pgrzesik** [comment]
Reference to docs about limitations sounds good to me @jugglingjsons :+1:

**Comment by jugglingjsons** [comment]
Updated, please check and verify. I've chosen to put it in one of the .md s only for now. The other seems to be a more general stack deployment guide.

File: docs/providers/aws/cli-reference/deploy-function.md [hunk 1]
```diff
@@ -23,7 +23,12 @@ serverless deploy function -f functionName
 **Note:** This command **now** deploys both function configuration and code by
 default. Just as before, this puts your function in an inconsistent state that
 is out of sync with your CloudFormation stack. Use this for faster development
-cycles and not production deployments
+cycles and not production deployments. Some parts of serverless configuration
```
Comment (pgrzesik) [inline docs/providers/aws/cli-reference/deploy-function.md:26]: Let's maybe rewrite it to something more like:

During `deploy function`, only changes that can be applied via `updateFunctionCode` and `updateFunctionConfiguration` direct `aws-sdk` calls will take effect. It is important to remember that configuration that is related to separate CloudFormation resources will not be updated. The function that is triggered through an alias will be updated, but the alias itself won't. One instance of that might be when using the `provisionedConcurrency`.

File: lib/plugins/deploy.js [hunk 2]
```diff
@@ -56,7 +56,8 @@ class Deploy {
         },
         commands: {
           function: {
-            usage: 'Deploy a single function from the service',
+            usage:
+              'Deploy a single function from the service. It has limitations - read more here: https://github.com/serverless/serverless/blob/master/docs/providers/aws/cli-reference/deploy-function.md',
```
Comment (pgrzesik) [inline lib/plugins/deploy.js:60]: After second though, I would remove that note. I've looked around and we do not practice linking to documentation in `usage` for specific commands + it's a general `function` command that is not only related to AWS provider, which will be confusing to people that use different providers.

**Review by pgrzesik — CHANGES_REQUESTED** [review]

**Comment by pgrzesik** [comment]
I'm going to close this one due to inactivity

**Pull request closed without merging**